crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -119,19 +161,20 @@ pub(crate) fn highlight(
 
```

Comment:
When we add highlights below this line, I think we have an invariant that the highlight must not extend beyond the current node (otherwise, flattening might be broken). Let's add an assert for this.